amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutConsumerRegistration.java

```diff
@@ -152,6 +152,12 @@ private void waitForActive() throws DependencyException {
         while (!ConsumerStatus.ACTIVE.equals(status) && retries > 0) {
             status = describeStreamConsumer().consumerDescription().consumerStatus();
             retries--;
+            try {
+                log.info(String.format("Waiting for StreamConsumer %s to have ACTIVE status...", streamConsumerName));
+                Thread.sleep(retryBackoffMillis);
+            } catch (InterruptedException ie) {
```

Comment:
This will cause the thread to continue processing even after being interrupted. A better behavior would be to stop the current processing and move on, this way the thread is not blocked.

```java
try {
    while(!ConsumerStatus.ACTIVE.equals(status) && retries > 0) {
        ...
    }
} catch (InterruptedException e) {
    log.debug("Thread was interrupted while fetching StreamConsumer status, moving on.");
}
```

Reply:
Good catch! Fixing :)